File: Postfix/Postfix.java
import java.util.ArrayDeque;
import java.util.Deque;
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        System.out.print("Enter a postfix expression: ");
        String exp = scanner.nextLine();
        String[] elems = exp.split(" ");

        Deque<String> stack = new ArrayDeque<String>();
        for (String elem : elems) {
            try {
                int num = Integer.parseInt(elem);
                stack.addFirst(Integer.toString(num));
            } catch(Exception e) {
                int b = Integer.parseInt(stack.removeFirst());
                int a = Integer.parseInt(stack.removeFirst());
                if (elem.equals("+")) stack.addFirst(Integer.toString(a + b));
                else if (elem.equals("-")) stack.addFirst(Integer.toString(a - b));
                else if (elem.equals("*")) stack.addFirst(Integer.toString(a * b));
                else if (elem.equals("/")) stack.addFirst(Integer.toString(a / b));
            }
        }
        System.out.println(stack.peek());
    }
}

File: Postfix/Postfix.cpp
#include <iostream>
#include <stack>
#include <sstream>

int main()
{
    std::cout << "Enter a postfix expression: ";
    std::string exp;
    getline(std::cin, exp);
    std::istringstream elems(exp);

    std::stack<std::string> stack;
    std::string elem;
    while (elems >> elem)
    {
        try
        {
            int num = stoi(elem);
            stack.push(std::to_string(num));
        }
        catch (...)
        {
            int b = stoi(stack.top());
            stack.pop();
            int a = stoi(stack.top());
            stack.pop();
            if (elem == "+") stack.push(std::to_string(a + b));
            else if (elem == "-") stack.push(std::to_string(a - b));
            else if (elem == "*") stack.push(std::to_string(a * b));
            else if (elem == "/") stack.push(std::to_string(a / b));
        }
    }
    std::cout << stack.top() << "\n";
}
